src/vaadin-grid-keyboard-navigation-mixin.html

```diff
@@ -515,7 +515,7 @@
 
     _resetKeyboardNavigation() {
       if (this.$.header.firstElementChild) {
-        this._headerFocusable = this.$.header.firstElementChild.firstElementChild;
+        this._headerFocusable = Array.from(this.$.header.firstElementChild.children).filter(el => !el.hidden)[0];
```

Comment:
Prior to filtering, it would be better to sort the columns by `column._order` so that the focusable is the _visually_ first cell in the row. Not a blocked for this PR as is still fixes the reported issue.